### PR TITLE
Make Gateways use Postgres Models

### PR DESCRIPTION
**Pull request by gcapalbo**
Probably best to review commit-by-commit, feel free to skip / gloss over the ones that mention "Copy code..." or "Move code..."

@proteusvacuum

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 15 errors:

* corehq/apps/sms/models.py, line 14 - F401 'Domain' imported but unused
* corehq/apps/sms/tests/test_backends.py, line 28 - F401 'json_format_datetime' imported but unused
* corehq/apps/sms/tests/test_backends.py, line 690 - E501 line too long (124 > 115 characters)
* corehq/apps/sms/tests/test_backends.py, line 699 - E501 line too long (117 > 115 characters)
* corehq/apps/sms/tests/test_backends.py, line 793 - F841 local variable 'backend_mapping1' is assigned to but never used
* corehq/apps/sms/tests/test_backends.py, line 800 - F841 local variable 'backend_mapping2' is assigned to but never used
* corehq/apps/sms/tests/test_backends.py, line 807 - F841 local variable 'backend_mapping3' is assigned to but never used
* corehq/apps/sms/tests/test_backends.py, line 815 - F841 local variable 'backend_mapping4' is assigned to but never used
* corehq/apps/sms/tests/test_backends.py, line 1085 - E303 too many blank lines (2)
* corehq/apps/sms/tests/test_registration.py, line 9 - F401 'settings' imported but unused
* corehq/apps/sms/tests/test_registration.py, line 10 - F401 'Site' imported but unused
* corehq/messaging/smsbackends/grapevine/models.py, line 140 - F841 local variable 'resp' is assigned to but never used
* corehq/messaging/smsbackends/http/models.py, line 221 - F841 local variable 'response' is assigned to but never used
* corehq/messaging/smsbackends/megamobile/models.py, line 124 - F841 local variable 'response' is assigned to but never used
* corehq/messaging/smsbackends/tropo/models.py, line 94 - E203 whitespace before ':'

**Comment by gcapalbo** [comment]
(Opening for review, hold off on merge for now, I want to run live tests with people Monday morning and then we should be good to go)

File: corehq/messaging/smsbackends/grapevine/models.py [hunk 1]
```diff
@@ -98,6 +98,47 @@ def get_available_extra_fields(cls):
...
+            affiliate_code=escape(config.affiliate_code),
+            auth_code=escape(config.authentication_code),
+            message=escape(text),
+            msisdn=escape(phone_number)
+        )
+
+        url = 'http://www.gvi.bms9.vine.co.za/httpInputhandler/ApplinkUpload'
+        req = urllib2.Request(url, data)
+        response = urllib2.urlopen(req, timeout=settings.SMS_GATEWAY_TIMEOUT)
+        resp = response.read()
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/grapevine/models.py]: F841 local variable 'resp' is assigned to but never used

File: corehq/messaging/smsbackends/http/models.py [hunk 2]
```diff
@@ -175,3 +175,51 @@ def get_available_extra_fields(cls):
...
+        params[config.message_param] = text
+        params[config.number_param] = phone_number
+
+        url_params = urlencode(params)
+        try:
+            if config.method == "GET":
+                response = urlopen("%s?%s" % (config.url, url_params),
+                    timeout=settings.SMS_GATEWAY_TIMEOUT).read()
+            else:
+                response = urlopen(config.url, url_params,
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/http/models.py]: F841 local variable 'response' is assigned to but never used

File: corehq/messaging/smsbackends/megamobile/models.py [hunk 3]
```diff
@@ -79,3 +79,46 @@ def get_available_extra_fields(cls):
...
+        config = self.config
+        params = urlencode({
+            "pid": pid,
+            "cel": phone_number,
+            "msg": text,
+            "src": config.source_identifier,
+        })
+        api_account_name = quote(config.api_account_name)
+        url = 'http://api.mymegamobile.com/%s?%s' % (api_account_name, params)
+        response = urlopen(url, timeout=settings.SMS_GATEWAY_TIMEOUT).read()
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/megamobile/models.py]: F841 local variable 'response' is assigned to but never used

**Comment by czue** [comment]
monster! :japanese_ogre:

**Comment by proteusvacuum** [comment]
This is huge! Things look good from me, although I don't really have enough context to comment on the content.

File: corehq/messaging/smsbackends/grapevine/models.py (hunk 1 above)
Comment (dimagimon) [inline corehq/messaging/smsbackends/grapevine/models.py]: F841 local variable 'resp' is assigned to but never used

File: corehq/messaging/smsbackends/http/models.py (hunk 2 above)
Comment (dimagimon) [inline corehq/messaging/smsbackends/http/models.py]: F841 local variable 'response' is assigned to but never used

File: corehq/messaging/smsbackends/megamobile/models.py (hunk 3 above)
Comment (dimagimon) [inline corehq/messaging/smsbackends/megamobile/models.py]: F841 local variable 'response' is assigned to but never used

**Comment by gcapalbo** [comment]
@nickpell can you review this commit: https://github.com/dimagi/commcare-hq/commit/990b1a5663815b045bd5e5a6cfb2d359199f98da?w=1 just want to make sure you're ok with that approach.  we no longer need split api ids in the gateway fee criteria (for example it used to be SISLOG api id for inbound, HTTP api id for outbound), so what this does is update the current gateway fee criteria records that use the HTTP api id with a migration, and also updates the earlier bootstrap code so that going forward a new bootstrap of fee criteria will use the right api ids the first time.

**Comment by gcapalbo** [comment]
@proteusvacuum updated, also I ran our gateways locally with live tests and we're good to go. as long as @nickpell is good with that last commit regarding the billable fee criteria update, then this should be good to merge

**Comment by proteusvacuum** [comment]
:+1:

**Comment by gcapalbo** [comment]
thanks @nickpell. @proteusvacuum we're good to merge now

**Comment by proteusvacuum** [comment]
:tada:

**Comment by czue** [comment]
holy :cow: this is huge! :fireworks:

**Pull request merged**